File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        str1 = ""
        str2 = ""
        current = head

        while current is not None:
            str1 += str(current.val)  # Append current value to str1
            str2 = str(current.val) + str2  # Prepend current value to str2
            current = current.next

        return str1 == str2

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function (head) {
    let arr = ''
    let a = ''
    let curr = head
    while (curr != null) {
        arr = arr + curr.val
        a = curr.val + a
        curr = curr.next
    }
    return arr == a

};